src/Windows/Avalonia.Win32/Input/WindowsInputPane.cs

```diff
@@ -84,9 +86,45 @@ public void Dispose()
         }
     }
 
-    private class Handler : CallbackBase, IFrameworkInputPaneHandler
+    private class Handler : IUnknown, IMicroComShadowContainer, IFrameworkInputPaneHandler
     {
-        private readonly WindowsInputPane _pane;
+        private WindowsInputPane _pane;
+
+        private readonly object _lock = new();
+        private bool _destroyed;
+        public bool IsDestroyed => _destroyed;
+
+        public void Dispose()
+        {
+            lock (_lock)
+            {
+                DestroyIfNeeded();
+            }
```

Comment:
It might be problematic to completely destroy this object, as it still might be used by native side.
To avoid Window memory leak, it will be enough to nullify "_pane" reference on disposal.

Reply:
I did as you suggest, `WindowsInputPane` is collected by GC but `WindowsInputPane+Handler` is reatain because `Shadow` keeps reference to it.

![immagine](https://github.com/user-attachments/assets/1d4ad326-9588-4121-aab5-9dc4eab8ec03)

![immagine](https://github.com/user-attachments/assets/ad15becb-6f2c-498d-8f07-a921ce804dc8)

I tried clear both references but WindowsInputPane+Handler is still active

```c#
    private class Handler : CallbackBase, IFrameworkInputPaneHandler
    {
        private WindowsInputPane _pane;

        public Handler(WindowsInputPane pane) => _pane = pane;
        public void Showing(UnmanagedMethods.RECT* rect, int _) => _pane.OnStateChanged(true, *rect);
        public void Hiding(int fEnsureFocusedElementInView) => _pane.OnStateChanged(false, null);

        public void Clear()
        {
            lock (this)
            {
                _pane = null!;
                Shadow = null;
            }
        }
    }
}
``` 

![immagine](https://github.com/user-attachments/assets/fd1c9edf-88f4-4d1c-a497-ae0c33ae5971)


Potentially all of the following classes are affected by the same problem

![immagine](https://github.com/user-attachments/assets/cb1b6eff-c84c-4660-9855-544232dd87af)


I think at this point that the best thing to do is to do two PRs

- one to change the behavior of `CallbackBase`
- the other to clear the reference to `WindowsInputPane`

@maxkatz6 , @kekekeks  What do you think?

Reply:
> Potentially all of the following classes are affected by the same problem

It needs to be confirmed whether it's MicroCOM holds these references when it shouldn't, or native side haven't released them. And I feel like it's the second case.
Either way, I don't think it's in the scope of this PR.

Just the fact that `Handler` doesn't leak whole Window with it is enough for this PR.